Login Rework: Enable SmartLock for Passwords in new login flow

hypest opened the pull request:
This PR enables the user of Google's SmartLock for Passwords in the new Login flows.

The user can save their WPCOM username+password pair in SmartLock and use it again later if logging into WPCOM is needed.

This PR enables saving the credentials only when using the username/password flow.

| Native Prompt for saved credentials |
| --- |
| ![prompt-for-saved-creds](https://user-images.githubusercontent.com/1032913/28621416-8a980d60-7219-11e7-9d38-970c1b1ea700.jpg) |

To test:
* With the app data cleaned, launch the app and hit "Log in"
* Tap on the "Log in to your site by entering your site address" secondary button
* Enter the site address *of your WPCOM* site. It's probably something like "<username>.wordpress.com" and hit "Next"
* Enter your username and password and hit "Next"
* Upon successful login, a popup will ask you to optionally save your credentials to SmartLock. Do so by tapping on "SAVE PASSWORD".
* At this point, the epilogue screen is visible. Ignore it and go clean the app data again to start over.
* Launch the app and hit the "Log in" button. A popup appears asking to choose a saved WordPress account from SmartLock. There must be at least one saved account, the one we saved at a previous step. Choose that account.
* The app shows the username+password screen in the background while attempting to login with the saved credentials. It should succeed and jump to the epilogue screen.

maxme commented:
Same kind of bug as above (Smart Lock dialog popping up when it should not):

* Tap Log In
* Tap "None of the above" in the Smart Lock Dialog
* Tap "Create New Acccount"
* Tap "Term of Services"
* Back
* 💥 Get the Smart Lock Dialog (shouldn't be there)
* 💥 Tap "None of the above" in the Smart Lock Dialog
* 💥 You get redirected to the Sign In screen

hypest commented:
I've addressed the crash and the SmartLock popup issue with 2d6fd97 and 338982d @maxme . While at it, I also fixed the weird styling issue with the "Create Account" screen (21a5534). Ready for another pass! Thanks!

maxme commented:
I confirm previous crash and bugs were fixed. 

:shipit: (but waiting for design review).

hypest commented:
Hey @maxme , i've piggybacked the fix for #6411 here. Please have a look, thanks!

folletto commented:
> Upon successful login, a popup will ask you to optionally save your credentials to SmartLock. Do so by tapping on "SAVE PASSWORD".

Didn't show up to me. Does it work with 2FA?

hypest commented:
> Does it work with 2FA?

Hey @folletto , no, as it currently is, it's only implemented in the username/password non-2FA flow. If you can use a non-2FA account it should work (provided SmartLock is also set up; enable it via https://passwords.google.com/).

I will work on enabling it on the 2FA flow as well.

maxme commented:
Fix for #6411  confirmed and 👍 for removing the error toast.

folletto commented:
I also had to turn on manually "Smart Lock for Passwords".

I was't able to test the whole sequence, but from the part I've seen and the screenshot above, combined with the fact it's a flow outside of our control, I'd say: let's merge. :)

hypest commented:
I enabled SmartLock password save in the email/password and 2FA flows as well (4ee0282) @folletto . /cc @maxme

hypest commented:
0347dec restricts the input chars for 2FA to simple numeric digits (no signs, period, or comma) @folletto. Thanks for the idea to limit it.

hypest commented:
@maxme the "needs design review" has been lifted but, a few new commits were added since the last time you reviewed. Ready for another pass. Thanks!

maxme commented:
Fix confirmed for #6111 and #6410, :shipit:
